src/picai_prep/dcm2mha.py

```diff
@@ -26,372 +29,389 @@
-        except Exception as e:
-            item['targets'][target] = f'Reading DICOM sequence failed, maybe corrupt data? Error: {e}'
-            self.item_log(item, 'corrupt data')
-            return False
 
-        if self.scan_postprocess_func is not None:
-            image = self.scan_postprocess_func(image)
+class Dicom2MHACase(Case):
+    settings: Dicom2MHASettings = Dicom2MHASettings({}, {})
 
-        try:
-            atomic_image_write(image=image, path=destination, mkdir=True)
-        except Exception as e:
-            item['targets'][target] = f'Error during writing: {e}'
-            self.item_log(item, 'write error')
-            return False
-        return True
+    def __init__(self, input_dir: Path, patient_id: str, study_id: str, paths: List[PathLike]):
+        self.patient_id = patient_id
+        self.study_id = study_id
+        self.series: List[Series] = []
+        self.log = [f'Importing {plural(len(paths), "serie")}']
 
-    def _convert(self):
-        convert_count = 0
-        for item in self.valid_items:
-            for _, status in item['targets'].items():
-                convert_count += 1 if status is None else 0
-
-        self.info(f"Converting {convert_count} DICOM series to MHA.")
-
-        # convert all items with valid targets to mha
-        success_count, skip_count = 0, 0
-        targets = []
-        for item in self.valid_items:
-            for target, status in item['targets'].items():
-                if status is None:  # therefore selected
-                    destination = os.path.join(self.output_dir, item['patient_id'], target + '.mha')
-                    if os.path.exists(destination):
-                        item['targets'][target] = 'Skipped, target output path already exists'
-                        self.item_log(item, 'skipped')
-                        skip_count += 1
-                    else:
-                        targets.append((item, target, destination))
+        fullpaths = set()
+        for path in paths:
+            full_path = input_dir / path
+            serie = Series(full_path, patient_id, study_id)
+            try:
+                if not full_path.exists():
+                    raise ArchiveItemPathNotFoundError(path)
+                elif not full_path.is_dir():
+                    raise NotADirectoryError(path)
+                elif path in fullpaths:
+                    raise FileExistsError(path)
+            except Exception as e:
+                serie.error = e
+                logging.error(str(e))
+            else:
+                fullpaths.add(full_path)
+            finally:
+                self.log.append(f'\t+ ({len(self.series)}) {full_path}')
+                self.series.append(serie)
+        if not all([serie.is_valid for serie in self.series]):
+            self.invalidate()
+
+    @property
+    def valid_series(self):
+        return [item for item in self.series if item.is_valid]
+
+    def invalidate(self):
```

Comment:
`valid_series` or `invalidate`?

Reply:
`invalidate`, not sure how GitHub decides which lines to show

Reply:
I use it as a clear STOP of the current faulty case. Because the program never stops, only error logs are displayed, time might be entirely wasted as valid series within a faulty case continue to convert rather uselessly. It's not super necessary, but I think its a nice sprinkle.